xds/internal/resolver/xds_resolver_test.go

```diff
@@ -147,7 +173,7 @@ func (s) TestResolverBuilder(t *testing.T) {
 				t.Fatalf("resolver.Get(%v) returned nil", xdsScheme)
 			}
 
-			r, err := builder.Build(target, newTestClientConn(), resolver.BuildOptions{})
+			r, err := builder.Build(test.target, newTestClientConn(), resolver.BuildOptions{})
```

Comment:
This comment is actually for `TestResolverBuilder_xdsCredsBootstrapMismatch`. We should probably change that to attempt to create a real xdsClient. Because this existing test clearly did not identify the bug where we could be leaking the xdsclient.
If you dont want to do this as part of this PR, even a TODO (with an issue) should be good.

Reply:
I refactored the tests to always check for client close.

Reply:
Thanks.